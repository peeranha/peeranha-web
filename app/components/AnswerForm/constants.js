export const TEXT_EDITOR_ANSWER_FORM =
  'app_ViewQuestion_TEXT_EDITOR_ANSWER_FORM';
