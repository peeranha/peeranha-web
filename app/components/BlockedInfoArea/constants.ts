export const BANNER_IMG = 'background_image';
export const ILLUSTRATION_IMG = 'illustration_image';
