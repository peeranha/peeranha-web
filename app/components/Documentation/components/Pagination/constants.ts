export const NEXT_TYPE_BUTTON = 'next';
export const PREV_TYPE_BUTTON = 'prev';
