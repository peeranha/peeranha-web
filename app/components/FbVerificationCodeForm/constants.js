export const VERIFY_FB_ACTION_FORM = 'VERIFY_FACEBOOK_ACTION_FORM';
export const FB_VERIFICATION_CODE_FIELD = 'FACEBOOK_VERIFICATION_CODE_FIELD';
