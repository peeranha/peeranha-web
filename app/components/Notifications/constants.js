import answerIcon from 'images/answer.svg?inline';
import questionIcon from 'images/question2.svg?inline';
import bestAnswerIcon from 'images/bestAnswer.svg?inline';
import downVoteIcon from 'images/redFingerDownSingleQuestion.svg?inline';
import upVoteIcon from 'images/fingerUpSingleQuestionPage.svg?inline';
import coinsIcon from 'images/coins.svg?inline';

import messages from './messages';

export const ROW_HEIGHT = 40;
export const VERTICAL_OFFSET = 5 * ROW_HEIGHT;
export const NOTIFICATIONS_REQUEST_LIMIT = 20;
export const MARK_AS_READ_DELAY = 1000;

export const MARK_ALL_NOTIFICATIONS_AS_READ =
  'app/Notifications/MARK_ALL_NOTIFICATIONS_AS_READ';
export const MARK_ALL_NOTIFICATIONS_AS_READ_SUCCESS =
  'app/Notifications/MARK_ALL_NOTIFICATIONS_AS_READ_SUCCESS';
export const MARK_ALL_NOTIFICATIONS_AS_READ_ERROR =
  'app/Notifications/MARK_ALL_NOTIFICATIONS_AS_READ_ERROR';

export const LOAD_MORE_UNREAD_NOTIFICATIONS =
  'app/Notifications/LOAD_MORE_UNREAD_NOTIFICATIONS';
export const LOAD_MORE_UNREAD_NOTIFICATIONS_SUCCESS =
  'app/Notifications/LOAD_MORE_UNREAD_NOTIFICATIONS_SUCCESS';
export const LOAD_MORE_UNREAD_NOTIFICATIONS_ERROR =
  'app/Notifications/LOAD_MORE_UNREAD_NOTIFICATIONS_ERROR';

export const LOAD_MORE_NOTIFICATIONS =
  'app/Notifications/LOAD_MORE_NOTIFICATIONS';
export const LOAD_MORE_NOTIFICATIONS_SUCCESS =
  'app/Notifications/LOAD_MORE_NOTIFICATIONS_SUCCESS';
export const LOAD_MORE_NOTIFICATIONS_ERROR =
  'app/Notifications/LOAD_MORE_NOTIFICATIONS_ERROR';

export const MARK_AS_READ_NOTIFICATIONS_ALL =
  'app/Notifications/MARK_AS_READ_NOTIFICATIONS_ALL';
export const MARK_AS_READ_NOTIFICATIONS_UNREAD =
  'app/Notifications/SET_READ_NOTIFICATIONS_UNREAD';

export const MARK_AS_READ_SUCCESS = 'app/Notifications/MARK_AS_READ_SUCCESS';
export const MARK_AS_READ_ERROR = 'app/Notifications/MARK_AS_READ_ERROR';

export const CLEAR_NOTIFICATIONS_DATA =
  'app/Notifications/CLEAR_NOTIFICATIONS_DATA';

export const SET_NOTIFICATIONS_INFO =
  'app/Notifications/SET_NOTIFICATIONS_INFO';

export const FILTER_UNREAD_TIMESTAMPS =
  'app/Notifications/FILTER_UNREAD_TIMESTAMPS';

export const NOTIFICATIONS_TYPES = {
  1: {
    ...messages.yourQuestionWasUpVoted,
    src: upVoteIcon,
  },
  2: {
    ...messages.yourQuestionWasDownVoted,
    src: downVoteIcon,
  },
  3: {
    ...messages.yourAnswerWasUpVoted,
    src: upVoteIcon,
  },
  4: {
    ...messages.yourAnswerWasDownVoted,
    src: downVoteIcon,
  },
  5: {
    ...messages.yourAnswerWasMarkedAsTheBest,
    src: bestAnswerIcon,
  },
  6: {
    ...messages.somebodyAnsweredYourQuestion,
    src: questionIcon,
  },
  7: {
    ...messages.somebodyLeftACommentToYourQuestion,
    src: questionIcon,
  },
  8: {
    ...messages.somebodyLeftACommentToYourAnswer,
    src: answerIcon,
  },
  9: {
    ...messages.yourQuestionWasTipped,
    src: coinsIcon,
  },
  10: {
    ...messages.yourAnswerWasTipped,
    src: coinsIcon,
  },
};
