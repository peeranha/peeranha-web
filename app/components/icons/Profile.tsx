import React from 'react';
import IconComponent, { IconProps } from './IconComponent';

const Profile: React.FC<IconProps> = (props): JSX.Element => (
  <IconComponent
    {...props}
    id="profile"
    fill="curentColor"
    viewBox="0 0 24 24"
    size={props.size}
  >
    <g fill="none">
      <circle
        cx="12"
        cy="7"
        r="5"
        fill="white"
        stroke={props.stroke || '#282828'}
      />
      <ellipse
        cx="12"
        cy="17"
        rx="8"
        ry="5"
        fill={props.fill || '#fff'}
        stroke={props.stroke || '#282828'}
      />
    </g>
  </IconComponent>
);

export default Profile;
