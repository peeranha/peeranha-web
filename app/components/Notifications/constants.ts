import * as routes from 'routes-config';

import { POST_TYPE } from 'utils/constants';

export const ROW_HEIGHT = 70;
export const ROW_HEIGHT_FOR_SMALL = 120;
export const NOTIFICATIONS_REQUEST_LIMIT = 50;
export const MARK_AS_READ_DELAY = 1000;

export const MARK_ALL_NOTIFICATIONS_AS_READ = 'app/Notifications/MARK_ALL_NOTIFICATIONS_AS_READ';
export const MARK_ALL_NOTIFICATIONS_AS_READ_SUCCESS =
  'app/Notifications/MARK_ALL_NOTIFICATIONS_AS_READ_SUCCESS';
export const MARK_ALL_NOTIFICATIONS_AS_READ_ERROR =
  'app/Notifications/MARK_ALL_NOTIFICATIONS_AS_READ_ERROR';

export const LOAD_MORE_UNREAD_NOTIFICATIONS = 'app/Notifications/LOAD_MORE_UNREAD_NOTIFICATIONS';
export const LOAD_MORE_UNREAD_NOTIFICATIONS_SUCCESS =
  'app/Notifications/LOAD_MORE_UNREAD_NOTIFICATIONS_SUCCESS';
export const LOAD_MORE_UNREAD_NOTIFICATIONS_ERROR =
  'app/Notifications/LOAD_MORE_UNREAD_NOTIFICATIONS_ERROR';

export const LOAD_MORE_NOTIFICATIONS = 'app/Notifications/LOAD_MORE_NOTIFICATIONS';
export const LOAD_MORE_NOTIFICATIONS_SUCCESS = 'app/Notifications/LOAD_MORE_NOTIFICATIONS_SUCCESS';
export const LOAD_MORE_NOTIFICATIONS_ERROR = 'app/Notifications/LOAD_MORE_NOTIFICATIONS_ERROR';

export const MARK_AS_READ_NOTIFICATIONS_ALL = 'app/Notifications/MARK_AS_READ_NOTIFICATIONS_ALL';
export const MARK_AS_READ_NOTIFICATIONS_UNREAD = 'app/Notifications/SET_READ_NOTIFICATIONS_UNREAD';

export const MARK_AS_READ_SUCCESS = 'app/Notifications/MARK_AS_READ_SUCCESS';
export const MARK_AS_READ_ERROR = 'app/Notifications/MARK_AS_READ_ERROR';

export const CLEAR_NOTIFICATIONS_DATA = 'app/Notifications/CLEAR_NOTIFICATIONS_DATA';

export const SET_NOTIFICATIONS_INFO = 'app/Notifications/SET_NOTIFICATIONS_INFO';

export const FILTER_READ_TIMESTAMPS = 'app/Notifications/FILTER_UNREAD_TIMESTAMPS';

export const SET_LAST_USER = 'app/Notifications/SET_LAST_USER';

export const NOTIFICATIONS_TYPES = {
  questionUpVoted: 1,
  questionDownVoted: 2,
  questionUpVoteCanceled: 3,
  questionDownVoteCanceled: 4,
  answerUpVoted: 5,
  answerDownVoted: 6,
  answerUpVoteCanceled: 7,
  answerDownVoteCanceled: 8,
  answerMarkedTheBest: 9,
  questionAnswered: 10,
  questionCommented: 11,
  answerCommented: 12,
  questionTipped: 13,
  answerTipped: 14,
  postTypeChanged: 15,
  communityChanged: 16,
};

export const NOTIFICATIONS_DATA = {
  [NOTIFICATIONS_TYPES.questionUpVoted]: {
    keyTranslate: 'notifications.yourQuestionWasUpVoted',
  },
  [NOTIFICATIONS_TYPES.questionDownVoted]: {
    keyTranslate: 'notifications.yourQuestionWasDownVoted',
  },
  [NOTIFICATIONS_TYPES.questionUpVoteCanceled]: {
    keyTranslate: 'notifications.yourQuestionUpVoteWasCanceled',
  },
  [NOTIFICATIONS_TYPES.questionDownVoteCanceled]: {
    keyTranslate: 'notifications.yourQuestionDownVoteWasCanceled',
  },
  [NOTIFICATIONS_TYPES.answerUpVoted]: {
    keyTranslate: 'notifications.yourAnswerWasUpVoted',
  },
  [NOTIFICATIONS_TYPES.answerDownVoted]: {
    keyTranslate: 'notifications.yourAnswerWasDownVoted',
  },
  [NOTIFICATIONS_TYPES.answerUpVoteCanceled]: {
    keyTranslate: 'notifications.yourAnswerUpVoteWasCanceled',
  },
  [NOTIFICATIONS_TYPES.answerDownVoteCanceled]: {
    keyTranslate: 'notifications.yourAnswerDownVoteWasCanceled',
  },
  [NOTIFICATIONS_TYPES.answerMarkedTheBest]: {
    keyTranslate: 'notifications.yourAnswerWasMarkedAsTheBest',
  },
  [NOTIFICATIONS_TYPES.questionAnswered]: {
    keyTranslate: 'notifications.somebodyAnsweredYourQuestion',
  },
  [NOTIFICATIONS_TYPES.questionCommented]: {
    keyTranslate: 'notifications.somebodyLeftACommentToYourQuestion',
  },
  [NOTIFICATIONS_TYPES.answerCommented]: {
    keyTranslate: 'notifications.somebodyLeftACommentToYourAnswer',
  },
  [NOTIFICATIONS_TYPES.questionTipped]: {
    keyTranslate: 'notifications.yourQuestionWasTipped',
  },
  [NOTIFICATIONS_TYPES.answerTipped]: {
    keyTranslate: 'notifications.yourAnswerWasTipped',
  },
  [NOTIFICATIONS_TYPES.postTypeChanged]: {
    keyTranslate: 'notifications.yourPostTypeWasChanged',
  },
  [NOTIFICATIONS_TYPES.communityChanged]: {
    keyTranslate: 'notifications.communityWasChanged',
  },
};

export const POST_TYPE_TO_LABEL: { [index: number]: string } = {
  0: 'common.expert',
  1: 'common.general',
  2: 'common.tutorial',
  3: 'createCommunity.discord',
  // To do: telegram and slack
  // 4: 'createCommunity.telegram',
  // 5: 'createCommunity.slack',
};

export const ROUTES_BY_TYPE = {
  [POST_TYPE.expertPost]: routes.expertPostView,
  [POST_TYPE.generalPost]: routes.questionView,
  [POST_TYPE.tutorial]: routes.tutorialView,
  [POST_TYPE.discord]: routes.discordPostView,
  // To do: telegram and slack
  // [POST_TYPE.telegram]: routes.telegramPostView,
  // [POST_TYPE.slack]: routes.slackPostView,
};
