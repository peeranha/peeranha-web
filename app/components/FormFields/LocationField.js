import React from 'react';
import PropTypes from 'prop-types';
import { LOCATION_FIELD } from 'containers/Profile/constants';

import LocationList from './LocationList';

export const cities = (list, sendProps) =>
  list.map(item => (
    <li
      className="cityItem"
      role="presentation"
      key={item.geonameId}
      onClick={() =>
        sendProps.chooseLocation(
          item.geonameId,
          `${item.name}, ${item.countryName}`,
        )
      }
    >{`${item.name}, ${item.countryName}`}</li>
  ));

export const CitiesList = (showLocationList, sendProps) =>
  showLocationList && sendProps.citiesList ? (
    <LocationList>{cities(sendProps.citiesList, sendProps)}</LocationList>
  ) : null;

function LocationField({ input, label, disabled, sendProps }) {
  const { ipfs } = sendProps.profile;
  const showLocationList =
    ipfs &&
    ipfs[LOCATION_FIELD] &&
    !ipfs[LOCATION_FIELD].id &&
    ipfs[LOCATION_FIELD].name;

  return (
    <div>
      <h6>{label}</h6>
      <div className="position-relative mb-0">
        <input
          disabled={disabled}
          {...input}
          autoComplete="off"
          placeholder={label}
          onChange={e => sendProps.getCitiesList(e.target.value)}
          value={ipfs && ipfs[LOCATION_FIELD] && ipfs[LOCATION_FIELD].name}
          className="form-control mb-0 location-field"
          type="text"
        />
        {CitiesList(showLocationList, sendProps)}
      </div>
    </div>
  );
}

LocationField.propTypes = {
  input: PropTypes.object,
  label: PropTypes.string,
  disabled: PropTypes.bool,
  sendProps: PropTypes.object,
};

export default LocationField;
