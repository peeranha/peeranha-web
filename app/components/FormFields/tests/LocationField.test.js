import { shallow } from 'enzyme';

import { LOCATION_FIELD } from 'containers/Profile/constants';
import LocationField, { cities, CitiesList } from '../LocationField';

describe('LocationField test', () => {
  it('test by snapshots', () => {
    const getCitiesList = jest.fn();
    const cmp = LocationField({
      input: {},
      label: 'string',
      sendProps: {
        citiesList: [{ geonameId: 1 }],
        getCitiesList,
        profile: {
          ipfs: {
            [LOCATION_FIELD]: {
              name: '',
            },
          },
        },
      },
    });

    expect(cmp).toMatchSnapshot();
    expect(getCitiesList).toHaveBeenCalledTimes(0);

    shallow(cmp)
      .find('.location-field')
      .simulate('change', { target: {} });

    expect(getCitiesList).toHaveBeenCalledTimes(1);
  });

  it('test cities', () => {
    const list = [{ geonameId: 1 }];
    const chooseLocation = jest.fn();
    const cmp = cities(list, { chooseLocation })[0];
    const wrapper = shallow(cmp);

    expect(wrapper).toMatchSnapshot();

    expect(chooseLocation).toHaveBeenCalledTimes(0);
    wrapper.find('.cityItem').simulate('click');
    expect(chooseLocation).toHaveBeenCalledTimes(1);
  });

  it('CitiesList', () => {
    const sendProps = {
      citiesList: [{ geonameId: 1 }],
      chooseLocation: jest.fn(),
    };
    expect(CitiesList(true, sendProps)).toMatchSnapshot();
  });
});
