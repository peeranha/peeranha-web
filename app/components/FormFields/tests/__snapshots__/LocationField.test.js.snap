// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationField test CitiesList 1`] = `
<LocationList__Wrapper>
  <li
    className="cityItem"
    onClick={[Function]}
    role="presentation"
  >
    undefined, undefined
  </li>
</LocationList__Wrapper>
`;

exports[`LocationField test test by snapshots 1`] = `
<div>
  <h6>
    string
  </h6>
  <div
    className="position-relative mb-0"
  >
    <input
      autoComplete="off"
      className="form-control mb-0 location-field"
      onChange={[Function]}
      placeholder="string"
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`LocationField test test cities 1`] = `
<li
  className="cityItem"
  key="1"
  onClick={[Function]}
  role="presentation"
>
  undefined, undefined
</li>
`;
